gemma/gemma-inl.h

```diff
@@ -312,21 +312,29 @@ class GemmaAttention {
                 const size_t interleaved_idx = task / layer_config_.kv_heads;
                 const size_t query_idx = interleaved_idx % num_queries_;
                 const size_t batch_idx = interleaved_idx / num_queries_;
-                const size_t pos = queries_pos_[query_idx] + batch_idx;
+                size_t pos = queries_pos_[query_idx] + batch_idx;
                 const size_t cache_pos = div_seq_len_.Remainder(pos);
                 const size_t kv_offset = cache_pos * cache_pos_size_ +
                                          layer_ * cache_layer_size_ +
                                          head * layer_config_.qkv_dim * 2;
                 KVCache& kv_cache = kv_caches_[query_idx];
+
+                const hwy::Divisor& div_grp_size { static_cast<uint32_t>(layer_config_.grp_size) };
```

Comment:
Nice, thanks for adding. The constructor of Divisor does a lot of work, so we want to hoist this out of the loops.
Maybe we can move it to a member of GemmaAttention?

Reply:
Fair point about the extra arguments. I am a bit more concerned about the optimizing, though, because this is a parallel section and I'm not sure how good the compiler is about hoisting outside of that.
Maybe as a compromise we could keep the hwy::Divisor a local variable, but outside/before the pool.Run?